app/api_alpha/views.py

```diff
@@ -555,7 +557,72 @@ def get_diff(request):
         return response
 
 
-class ContributionsViewSet(viewsets.ModelViewSet):
+class ContributionsViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
     queryset = Contribution.objects.all()
-    http_method_names = ["post"]
     serializer_class = ContributionSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = ContributionSerializer(data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            if request.query_params.get("classement") == "true" and request.data.get(
+                "email"
+            ):
+                count, rank = get_contributor_count_and_rank(request.data.get("email"))
+                response = dict(serializer.data)
+                response["contributor_count"] = count
+                response["contributor_rank"] = rank
+                return Response(response, status=201)
+            else:
+                return Response(serializer.data, status=201)
+        else:
+            return Response(serializer.errors, status=400)
+
+    @action(detail=False, methods=["get"])
+    def classement(self, request, *args, **kwargs):
```

Comment:
Idem. Restons en anglais non ?
```suggestion
    def ranking(self, request, *args, **kwargs):
```

Reply:
ok